docs/external-generated/packages/powerpack-conformance/documents/create-conformance-rule.md

```diff
@@ -0,0 +1,308 @@
+
+Therefore, in practice, writing your local conformance rules in an Nx generated library is the easiest way to organize them and ensure that they are easily resolvable via TypeScript. The library in question could also be an Nx plugin, but it does not have to be.
+
+To write your own conformance rule, run the `@nx/powerpack-conformance:create-rule` generator and answer the prompts.
+
+```text {% command="nx g @nx/powerpack-conformance:create-rule" %}
+ NX  Generating @nx/powerpack-conformance:create-rule
+
+✔ What is the name of the rule? · local-conformance-rule-example
+✔ Which directory do you want to create the rule directory in? · packages/my-plugin/local-conformance-rule
+✔ What category does this rule belong to? · security
+✔ What reporter do you want to use for this rule? · project-reporter
+✔ What is the description of the rule? · an example of a conformance rule
+CREATE packages/my-plugin/local-conformance-rule/local-conformance-rule-example/index.ts
+CREATE packages/my-plugin/local-conformance-rule/local-conformance-rule-example/schema.json
+```
+
+The generated rule definition file should look like this:
+
+```ts {% fileName="packages/my-plugin/local-conformance-rule/index.ts" %}
+import {
+  createConformanceRule,
+  ProjectViolation,
+} from '@nx/powerpack-conformance';
+
+export default createConformanceRule({
+  name: 'local-conformance-rule-example',
+  category: 'security',
+  description: 'an example of a conformance rule',
+  reporter: 'project-reporter',
+  implementation: async (context) => {
+    const violations: ProjectViolation[] = [];
+
+    return {
+      severity: 'low',
+      details: {
+        violations,
+      },
+    };
+  },
+});
+```
+
+To enable the rule, you need to register it in the `nx.json` file.
+
+```json {% fileName="nx.json" %}
+{
+  "conformance": {
+    "rules": [
+      {
+        "rule": "./packages/my-plugin/local-conformance-rule/index.ts"
+      }
+    ]
+  }
+}
+```
+
+Note that the severity of the error is defined by the rule author and can be adjusted based on the specific violations that are found.
+
+## Conformance Rule Examples
+
+There are three types of reporters that a rule can use.
+
+- `project-reporter` - The rule evaluates an entire project at a time.
+- `project-files-reporter` - The rule evaluates a single project file at a time.
+- `non-project-files-reporter` - The rule evaluates files that don't belong to any project.
+
+{% tabs %}
+{% tab label="project-reporter" %}
+
+The `@nx/powerpack-owners` plugin adds an `owners` metadata property to every project node that has an owner in the project graph. This rule checks each project node metadata to make sure that each project has some owner defined.
+
+```ts
+import { ProjectGraphProjectNode } from '@nx/devkit';
+import {
+  createConformanceRule,
+  ProjectViolation,
+} from '@nx/powerpack-conformance';
+
+export default createConformanceRule({
+  name: 'ensure-owners',
```

Comment:
This rule is exposed via the conformance package already, so we shouldn't encourage folks to reimplement it directly without clarifying it's just for illustrative purposes